client/repolist.c

```diff
@@ -609,18 +609,23 @@ TDNFRepoListFinalize(
         BAIL_ON_TDNF_ERROR(dwError);
     }
 
-    //There could be overrides to enable/disable
-    //repo such as cmdline args, api overrides
-    pSetOpt = pTdnf->pArgs->pSetOpt;
-
-    while(pSetOpt)
+    /* There could be overrides to enable/disable
+       repo such as cmdline args, api overrides */
+    for (pSetOpt = pTdnf->pArgs->pSetOpt; pSetOpt; pSetOpt = pSetOpt->pNext)
     {
-        if(pSetOpt->nType == CMDOPT_ENABLEREPO ||
-           pSetOpt->nType == CMDOPT_DISABLEREPO)
+        if(strcmp(pSetOpt->pszOptName, "enablerepo") == 0)
+        {
+            dwError = TDNFAlterRepoState(
+                          pTdnf->pRepos,
+                          1,
+                          pSetOpt->pszOptValue);
+            BAIL_ON_TDNF_ERROR(dwError);
```

Comment:
put this BAIL_ON_TDNF_ERROR at the end of loop after all if else conditions, it is common in all cases

Reply:
Good point here.